Controller/HealthCheckController.php

```diff
@@ -146,10 +178,43 @@ protected function runTests(Request $request, $checkId = null)
         }
 
         $reporter = new ArrayReporter();
-        $this->runner->addReporter($reporter);
-        $this->runner->useAdditionalReporters($reporters);
-        $this->runner->run($checkId);
+
+        $runner = $this->getRunner($request);
+
+        $runner->addReporter($reporter);
+        $runner->useAdditionalReporters($reporters);
+        $runner->run($checkId);
 
         return $reporter;
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return Runner
+     *
+     * @throws \Exception
+     */
+    private function getRunner(Request $request)
+    {
+        $group = $this->getGroup($request);
+
+        $runner = $this->runnerManager->getRunner($group);
+
+        if ($runner) {
+            return $runner;
+        }
+
+        throw new \RuntimeException(sprintf('Unknown check group "%s"', $group));
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getGroup(Request $request)
+    {
+        return $request->query->get('group') ?: $this->runnerManager->getDefaultGroup();
```

Comment:
no my concern is not with group name 0 .. but more I was wondering why ``$this->runnerManager->getRunner(null)`` not simply assume default group, ie. move the entire handling of the default group to the runner manager.